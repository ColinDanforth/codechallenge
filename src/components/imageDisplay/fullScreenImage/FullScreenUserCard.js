import React from 'react'
import PropTypes from 'prop-types'
import UserDataCard from "../../imageCommon/UserDataCard"
import CameraDetailDataCard from "../../imageCommon/CameraDetailsDataCard"
import Title from "../../imageCommon/Title"

const rootDiv={
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'flex-start',
  height: 'auto',
  marginTop: '1%',
  marginRight: '2%',
  marginLeft: '2%',
}

const dataCardStyle={
  maxWidth: '820px',
  minWidth: '820px',
  width: 'auto',
  height:'auto',
}

const cameraDetailStyle={
  maxWidth: '820px',
  minWidth: '820px',
  width: 'auto',
  height:'auto',
  marginBottom: '4%',
}

const titleDiv = {
  maxWidth: '100vw',
  width: 'auto',
  marginBottom: '2%'
}

const FullScreenUserCard = (props)  => {
  return(
    <div style={rootDiv}>
      <div style={titleDiv}>
        <Title
          name={props.dataObject.name}
          description={props.thisItem.description}
        />
      </div>
      <div style={dataCardStyle}>
        <UserDataCard dataObject={props.dataObject}/>
      </div>
      <div style={cameraDetailStyle}>
        <CameraDetailDataCard dataObject={props.dataObject}/>
      </div>
    </div>
  )
}

export default FullScreenUserCard

FullScreenUserCard.propTypes={
  dataObject: PropTypes.object,
  thisItem: PropTypes.object,
}
