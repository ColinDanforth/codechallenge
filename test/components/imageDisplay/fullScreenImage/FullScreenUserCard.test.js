import React from 'react';
import FullScreenUserCard from '../../../../src/components/imageDisplay/fullScreenImage/FullScreenUserCard';
import renderer from 'react-test-renderer';

test('FullScreenUserCard should display a group of comment elements showing the appropriate text from the testData object', () => {
  const testDataObject={
    name: 'test name',
    user: {
      avatars: {
        default: {
          https: 'http://google.ca'
        }
      },
      fullname: 'Test User',

    },
    rating: '99.99999995',
    lens_info: {
      friendly_name: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
      features: "Micro Four Thirds System<br/>24-80mm (35mm Equivalent)<br/>Aperture Range: f/2.8 to f/22",
    },
    camera: "Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO",
    shutter_speed: 'test string',
    aperture: 'test string',
  }

  const testItem = {
    description: 'test description',
  }

  const component = renderer.create(
    <FullScreenUserCard dataObject={testDataObject} thisItem={testItem}/>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})
