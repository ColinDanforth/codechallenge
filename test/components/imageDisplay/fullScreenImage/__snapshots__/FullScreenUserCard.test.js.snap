// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullScreenUserCard should display a group of comment elements showing the appropriate text from the testData object 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
      "flexDirection": "column",
      "height": "auto",
      "justifyContent": "space-between",
      "width": "auto",
    }
  }
>
  <div
    style={
      Object {
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "height": "auto",
        "marginBottom": "2%",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderBottom": "inset",
          "display": "flex",
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "flexDirection": "Column",
          "height": "auto",
          "justifyContent": "center",
          "minWidth": "40vw",
          "width": "auto",
        }
      }
    >
      <h1>
        test name
      </h1>
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "test description",
          }
        }
        style={
          Object {
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "height": "auto",
            "margin": "3%",
            "width": "90%",
          }
        }
      />
    </div>
  </div>
  <div
    style={
      Object {
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "height": "auto",
        "marginBottom": "2%",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderBottom": "inset",
          "borderRight": "inset",
          "display": "flex",
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "flexDirection": "column",
          "height": "auto",
          "justifyContent": "flex-start",
          "minWidth": "100%",
          "width": "auto",
        }
      }
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "flexDirection": "row",
            "height": "auto",
            "justifyContent": "flex-start",
            "minWidth": "100%",
            "width": "auto",
          }
        }
      >
        <img
          alt="avatar"
          src="http://google.ca"
          style={
            Object {
              "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
              "height": "auto",
              "margin": "0.3%",
              "maxHeight": "120px",
              "width": "auto",
            }
          }
        />
        <div
          style={
            Object {
              "display": "flex",
              "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
              "flexDirection": "row",
              "height": "auto",
              "justifyContent": "space-between",
              "marginLeft": "1%",
              "minWidth": "78%",
              "width": "auto",
            }
          }
        >
          <div
            style={
              Object {
                "display": "flex",
                "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                "flexDirection": "column",
                "height": "auto",
                "justifyContent": "flex-start",
                "minWidth": "85%",
                "width": "auto",
              }
            }
          >
            <p
              style={
                Object {
                  "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                  "height": "auto",
                  "width": "auto",
                }
              }
            >
              Photographer: 
              Test User
            </p>
            <p
              style={
                Object {
                  "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                  "height": "auto",
                  "width": "auto",
                }
              }
            >
              Rating: 
              99.99999995
              %
            </p>
          </div>
          <div
            style={
              Object {
                "alignSelf": "right",
                "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                "height": "auto",
                "marginTop": "5px",
                "maxHeight": "40px",
                "width": "auto",
              }
            }
          >
            <img
              alt="like"
              src="test-file-stub"
              style={
                Object {
                  "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
                  "height": "auto",
                  "maxWidth": "40px",
                  "width": "auto",
                }
              }
            />
          </div>
        </div>
      </div>
      <div
        style={
          Object {
            "alignItems": "left",
            "display": "flex",
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "flexDirection": "column",
            "height": "auto",
            "justifyContent": "flex-start",
            "marginTop": "1%",
            "minWidth": "100%",
            "width": "auto",
          }
        }
      >
        <p
          style={
            Object {
              "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
              "marginLeft": "2%",
              "marginRight": "2%",
              "marginTop": "1%",
            }
          }
        >
          Location: 
          Toronto
          , 
          Canada
        </p>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
        "height": "auto",
        "marginBottom": "2%",
        "width": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundColor": "RGBA(256,256,256,1)",
          "borderBottom": "inset",
          "display": "flex",
          "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginTop": "1%",
        }
      }
    >
      <p
        style={
          Object {
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "marginLeft": "2%",
            "marginRight": "2%",
            "marginTop": "1%",
          }
        }
      >
        Camera: 
        Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
      </p>
      <p
        style={
          Object {
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "marginLeft": "2%",
            "marginRight": "2%",
            "marginTop": "1%",
          }
        }
      >
        Shutter Speed: 
        test string
      </p>
      <p
        style={
          Object {
            "flex": "none | [ <'flex-grow'> <'flex-shrink'>? || <'flex-basis'> ]",
            "marginLeft": "2%",
            "marginRight": "2%",
            "marginTop": "1%",
          }
        }
      >
        Aperture: 
        test string
      </p>
      <div>
        <p
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "30px",
              "marginRight": "30px",
            }
          }
        >
          Lens Info:
        </p>
        <p
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "60px",
              "marginRight": "30px",
            }
          }
        >
          Name: 
          Olympus M.Zuiko Digital ED 12-40mm f/2.8 PRO
        </p>
        <p
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "60px",
              "marginRight": "30px",
            }
          }
        >
          Feature: 
          Micro Four Thirds System
        </p>
        <p
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "60px",
              "marginRight": "30px",
            }
          }
        >
          24-80mm (35mm Equivalent)
        </p>
        <p
          style={
            Object {
              "marginBottom": "15px",
              "marginLeft": "60px",
              "marginRight": "30px",
            }
          }
        >
          Aperture Range: f/2.8 to f/22
        </p>
      </div>
    </div>
  </div>
</div>
`;
